### ISSUE TITLE
ModuleInitializerAttribute source generated, causing conflict

### ISSUE THREAD

**Issue by AArnott** [thomhurst/TUnit#1464]
[You predicted it](https://github.com/thomhurst/TUnit/pull/1436#issuecomment-2561947101), and it happened. I'm using "PolySharp" in my test project already and it's generating the `ModuleInitializerAttribute` for me, but then your TUnit.Core.SourceGenerator comes along and emits it as well, causing a CS0101 error:

> The namespace 'System.Runtime.CompilerServices' already contains a definition for 'ModuleInitializerAttribute'

Is there a way to turn yours off? (I thought you'd decided to not emit this, and rather to document that users may need to reference or declare a polyfill themselves).

**Comment by AArnott** [comment]
I was able to workaround this by defining my own (3rd, ironically) polyfill as an ordinary source file:

```cs
namespace System.Runtime.CompilerServices;

[AttributeUsage(AttributeTargets.Method, Inherited = false)]
internal sealed class ModuleInitializerAttribute : Attribute;
```

This gets both source generators to stop emitting it, avoiding the collision.

Still, it adds friction (and the workaround isn't particularly discoverable if you're not fluent in how source generators may work) to migrating to TUnit.

**Comment by thomhurst** [comment]
Maybe adding a project property would be useful to disable it?

`<EnableTUnitPolyfills>false</EnableTUnitPolyfills>`

**Comment by AArnott** [comment]
Your fix for this is ineffective, I'm afraid. I'm setting `EnableTUnitPolyfills` to `false` and your source generator still emits polyfills.

While you check within your source generator for the `EnableTUnitPolyfills` property, you don't have access to it unless your nuget package carries a .props or .targets file that has this item:

```xml
   <CompilerVisibleProperty Include="EnableTUnitPolyfills" />
```

@thomhurst can you reactivate?

**Comment by thomhurst** [comment]
Thanks @AArnott I didn't know about that property

**Comment by thomhurst** [comment]
@AArnott can you try on the latest version?

**Comment by AArnott** [comment]
Yes, 0.21.13 allows me to remove my workaround and just set the project property. Thanks.

**Comment by iikuzmychov** [comment]
just noticed the same issue with v1.3.25. A single workaround which works for me is the mentioned one:
```csharp
namespace System.Runtime.CompilerServices;

[AttributeUsage(AttributeTargets.Method, Inherited = false)]
internal sealed class ModuleInitializerAttribute : Attribute;
```

I think that if you need to use pollyfils somewhere in TUnit, I would suggest to try [PolySharp](https://github.com/Sergio0694/PolySharp), works like a charm
